ingest/src/main/java/com/microsoft/azure/kusto/ingest/resources/RankedStorageAccount.java

```diff
@@ -82,8 +83,9 @@ public double getRank() {
         // For each bucket, calculate the success rate ( success / total ) and multiply it by the bucket weight.
         // The older the bucket, the less weight it has. For example, if there are 3 buckets, the oldest bucket will have
         // a weight of 1, the middle bucket will have a weight of 2 and the newest bucket will have a weight of 3.
-
-        for (Bucket bucket : buckets) {
+        Iterator<Bucket> iterator = buckets.iterator();
```

Comment:
We could alternatively lock this section from read access while `buckets` is being modified, but I don't think that's necessary, and therefore not worth the perf penalty.